proposals/records.md

```diff
@@ -84,19 +94,29 @@ constructor and a constructor with the same signature already present in the cla
 
 At runtime the primary constructor
 
-1. executes the instance field initializers appearing in the class-body; and then
-    invokes the base class constructor with no arguments.
+1. executes the instance initializers appearing in the class-body
+
+1. invokes the base class constructor with the arguments provided in the `record_base` clause, if present
 
-1. initializes compiler-generated backing fields for the properties corresponding to the value parameters (if these properties are compiler-provided
 
 ### Properties
 
-For each record parameter of a record type declaration there is a corresponding public property member whose name and type are taken from the value parameter declaration. If no concrete (i.e. non-abstract) property with a get accessor and with this name and type is explicitly declared or inherited, it is produced by the compiler as follows:
+For each record parameter of a record type declaration there is a corresponding public property
+member whose name and type are taken from the value parameter declaration.
 
-For a record struct or a record class:
+For a record:
 
-* A public `get` and `init` auto-property is created (see separate `init` accessor specification). Its value is initialized during construction with the value of the corresponding primary constructor parameter. Each "matching" inherited abstract property's get accessor is overridden.
+* A public `get` and `init` auto-property is created (see separate `init` accessor specification).
+  Each "matching" inherited abstract accessor is overridden. The auto-property is initialized to
+  the value of the corresponding primary constructor parameter.
 
+### Deconstruct
+
+A positional record synthesizes a public void-returning method called Deconstruct with an out
+parameter declaration for each parameter of the primary constructor declaration. Each parameter
+of the Deconstruct method has the same type as the corresponding parameter of the primary
+constructor declaration. The body of the method assigns each parameter of the Deconstruct method
+to the value from an instance member access to a member of the same name.
```

Comment:
What happens if there is a positional parameter with the same name as a property or field that is already present in the type? e.g.

```cs
record C(int X)
{
    public int X { get; init; }
}
```

It feels like we *shouldn't* generate a Deconstruct method in this case, for the same reason we don't automatically assign from the positional parameter to the explicitly declared property.

Reply:
Does that mean that patterns like `case Some(var value)` would not be possible?

Reply:
Not sure I completely follow, but users are free to write whatever Deconstruct method is appropriate for their type. The compiler should refrain from generating the Deconstruct method if a method with the same signature exists already on the record type. Not sure what should happen if the Deconstruct() method is on the base type. I don't think that is specified here, so let's make sure we come to a conclusion on that question.

Reply:
I'm sorry, I read that as not generating a `Deconstruct` for a positional record with a single parameter.  Nevermind.

Reply:
I think we should generate a Deconstruct regardless. If the user doesn't want one, they can declare a private method with the same signature.